priv/templates/extended_bin

```diff
@@ -173,60 +179,59 @@ relx_rem_sh() {
     id="remsh$(relx_gen_id)-${NAME}"
 
     # Get the node's ticktime so that we use the same thing.
-    TICKTIME="$(relx_nodetool rpcterms net_kernel get_net_ticktime)"
+    TICKTIME="$(erl_call 'net_kernel get_net_ticktime')"
 
     # Setup remote shell command to control node
+    # -dist_listen is new in OTP-23. It keeps the remote node from binding to a listen port
+    # and implies the optoin -hidden
     exec "$BINDIR/erl" "$NAME_TYPE" "$id" -remsh "$NAME" -boot "$REL_DIR"/start_clean -mode interactive \
-         -boot_var SYSTEM_LIB_DIR "$SYSTEM_LIB_DIR" $MAYBE_DIST_ARGS \
-         -setcookie "$COOKIE" -hidden -kernel net_ticktime $TICKTIME
+         -boot_var SYSTEM_LIB_DIR "$SYSTEM_LIB_DIR" \
+         -setcookie "$COOKIE" -hidden -kernel net_ticktime "$TICKTIME" \
+         -dist_listen false \
+         "$MAYBE_DIST_ARGS"
 }
 
-# Generate a random id
-relx_gen_id() {
-    od -t x -N 4 /dev/urandom | head -n1 | awk '{print $2}'
+erl_call() {
+    command=$*
+
+    "$ROOTDIR/bin/erl_call" "$NAME_TYPE" "$NAME" -c "${COOKIE}" -a "${command}"
```

Comment:
This works on linux, but somehow on OSX, I didn't get erl_call in a release's rootdir, but always in `lib/erl_interface-<vsn>/bin/erl_call` and my path doesn't resolve there by default like it does on linux.

After digging it seems that this was due to having an old kerl version (1.3.2, from 2016) on OSX compared to Linux (1.8.6).

Still, I also don't see any code in relx master that moves or brings that executable outside of that directory, and not sure how the $ROOTDIR stuff here should hold up?

Reply:
Interesting... It moves it in `rlx_assemble`:

```
            %% erl_call is a bin in erl_interface that we must make a copy of
            %% so it is usable in release scripts from the release main bin dir
            Bin = code:lib_dir(erl_interface, bin),
            ErlCall = filename:join(Bin, "erl_call"),
            BinDir = filename:join([OutputDir, "bin"]),
            LocalErlCall = filename:join(BinDir, "erl_call"),
            ec_file:copy(ErlCall, LocalErlCall),
```

Reply:
Oh, I just realized you meant `erl_call` wasn't in your `PATH`. I thought you meant that the `ec_file:copy` was putting it in `bin/lib/erl_interface-<vsn>/bin/erl_call` in the release somehow.

Today I'm going to fix it so it only copies it if erts is included and if `$ROOT/bin/erl_call` isn't found it will either look in the path or use like:

```
erl_call="$("$__erl" -boot no_dot_erlang -sasl errlog_type error -noshell -eval "code:lib_dir(erl_interface, bin)")"
```

I originally was only going to put in the latter and hadn't considered the fact that it might make more sense to just use `PATH`... So unsure if the latter should even be included in the script.

Reply:
I think my grepping failed me on the file copy. It's better to have it in the release than $PATH -- I won't have it in $PATH In a docker image.